File: InPlan.Web/Modules/Tenencia/Areas/AreasForm.cs
using Serenity.ComponentModel;

namespace InPlan.Tenencia.Forms;

[FormScript("Tenencia.Areas")]
[BasedOnRow(typeof(AreasRow), CheckNames = true)]
public class AreasForm
{
    [HalfWidth]
    public int IdEmpresa { get; set; }
    [HalfWidth]
    public string Descripcion { get; set; }
    [TextAreaEditor(Rows = 5)]
    public string Observaciones { get; set; }
    [HideOnInsert]
    public bool Baja { get; set; }
}

File: InPlan.Web/Modules/ServerTypes/Tenencia/AreasForm.ts
import { LookupEditor, StringEditor, TextAreaEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AreasForm {
    IdEmpresa: LookupEditor;
    Descripcion: StringEditor;
    Observaciones: TextAreaEditor;
    Baja: BooleanEditor;
}

export class AreasForm extends PrefixedContext {
    static formKey = 'Tenencia.Areas';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AreasForm.init)  {
            AreasForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;
            var w3 = BooleanEditor;

            initFormType(AreasForm, [
                'IdEmpresa', w0,
                'Descripcion', w1,
                'Observaciones', w2,
                'Baja', w3
            ]);
        }
    }
}